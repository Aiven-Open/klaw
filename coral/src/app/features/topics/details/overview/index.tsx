import { useTopicDetails } from "src/app/features/topics/details/TopicDetails";

function TopicOverview() {
  const { topicName } = useTopicDetails();

  return <div>Placeholder for Overview panel, topic name: {topicName}</div>;
}

export { TopicOverview };
