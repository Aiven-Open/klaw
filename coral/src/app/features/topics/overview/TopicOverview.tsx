import { Navigate, useMatches } from "react-router-dom";
import {
  isTopicsOverviewTabEnum,
  TOPIC_OVERVIEW_TAB_ID_INTO_PATH,
  TopicOverviewTabEnum,
} from "src/app/router_utils";
import { TopicOverviewResourcesTabs } from "src/app/features/topics/overview/components/TopicOverviewResourceTabs";
import { TopicOverviewHeader } from "src/app/features/topics/overview/components/TopicOverviewHeader";

type TopicOverviewProps = {
  topicName: string;
};

function findMatchingTab(
  matches: ReturnType<typeof useMatches>
): TopicOverviewTabEnum | undefined {
  const match = matches
    .map((match) => match.id)
    .find((id) =>
      Object.prototype.hasOwnProperty.call(TOPIC_OVERVIEW_TAB_ID_INTO_PATH, id)
    );
  if (isTopicsOverviewTabEnum(match)) {
    return match;
  }
  return undefined;
}

function TopicOverview(props: TopicOverviewProps) {
  const { topicName } = props;

  const matches = useMatches();
  const currentTab = findMatchingTab(matches);
  if (currentTab === undefined) {
    return <Navigate to={`/topics`} replace={true} />;
  }
  return (
    <div>
      <TopicOverviewHeader topicName={topicName} />

      <TopicOverviewResourcesTabs
        currentTab={currentTab}
        topicName={topicName}
      />
    </div>
  );
}

export { TopicOverview };
