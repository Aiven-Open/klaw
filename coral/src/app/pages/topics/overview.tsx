function TopicOverviewPage() {
  return <div>{"<TopicOverview /> goes here"}</div>;
}

export { TopicOverviewPage };
