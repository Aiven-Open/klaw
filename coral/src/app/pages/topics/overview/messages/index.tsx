import { TopicMessages } from "src/app/features/topics/overview/messages/TopicMessages";

function TopicMessagesPage() {
  return <TopicMessages />;
}

export { TopicMessagesPage };
