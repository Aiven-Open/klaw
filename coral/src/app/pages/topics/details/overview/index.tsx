import { TopicOverview } from "src/app/features/topics/details/overview";

function TopicOverviewPage() {
  return <TopicOverview />;
}

export { TopicOverviewPage };
