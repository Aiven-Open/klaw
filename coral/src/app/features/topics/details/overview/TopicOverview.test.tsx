import { TopicOverview } from "src/app/features/topics/details/overview/TopicOverview";
import { customRender } from "src/services/test-utils/render-with-wrappers";
import {
  ClusterDetails as ClusterDetailsType,
  getClusterDetails,
} from "src/domain/cluster";
import { cleanup, RenderResult, screen, within } from "@testing-library/react";
import { getDefinitionList } from "src/services/test-utils/custom-queries";

const mockedUseTopicDetails = jest.fn();
jest.mock("src/app/features/topics/details/TopicDetails", () => ({
  useTopicDetails: () => mockedUseTopicDetails(),
}));

jest.mock("src/domain/cluster/cluster-api.ts");

const mockGetClusterDetails = getClusterDetails as jest.MockedFunction<
  typeof getClusterDetails
>;

const mockUseTopicDetailsDataWithPromotion = {
  topicName: "aivendemotopic",
  environmentId: "1",
  topicOverview: {
    topicExists: true,
    schemaExists: false,
    topicInfo: {
      topicName: "aivendemotopic",
      noOfPartitions: 1,
      noOfReplicas: "1",
      teamname: "Ospo",
      teamId: 1003,
      envId: "1",
      showEditTopic: true,
      showDeleteTopic: false,
      topicDeletable: false,
      envName: "DEV",
      topicOwner: true,
    },
    aclInfoList: [
      {
        req_no: "1061",
        aclPatternType: "LITERAL",
        environment: "1",
        environmentName: "DEV",
        kafkaFlavorType: "AIVEN_FOR_APACHE_KAFKA",
        showDeleteAcl: true,
        teamid: 1003,
        teamname: "Ospo",
        topicname: "aivendemotopic",
        topictype: "Producer",
        acl_ssl: "aivendemotopicuser",
        consumergroup: "-na-",
      },
    ],
    topicHistoryList: [
      {
        environmentName: "DEV",
        teamName: "Ospo",
        requestedBy: "muralibasani",
        requestedTime: "2023-May-17 10:13:47",
        approvedBy: "josepprat",
        approvedTime: "2023-May-17 10:14:01",
        remarks: "Create",
      },
    ],
    availableEnvironments: [
      {
        id: "1",
        name: "DEV",
      },
      {
        id: "2",
        name: "TST",
      },
    ],
    topicPromotionDetails: {
      status: "SUCCESS",
      sourceEnv: "1",
      targetEnv: "TST",
      targetEnvId: "2",
      topicName: "aivendemotopic",
    },
  },
  topicSchemas: {
    topicExists: true,
    schemaExists: true,

    allSchemaVersions: {
      DEV: [1],
    },
    latestVersion: {
      DEV: 1,
    },
    schemaPromotionDetails: {
      DEV: {
        status: "SUCCESS",
        sourceEnv: "3",
        targetEnv: "TST_SCH",
        targetEnvId: "9",
      },
    },
    schemaDetails: [
      {
        id: 2,
        version: 1,
        nextVersion: 0,
        prevVersion: 0,
        compatibility: "BACKWARD",
        content:
          '{\n  "doc" : "example",\n  "fields" : [ {\n    "default" : "6666665",\n    "doc" : "my test number",\n    "name" : "test",\n    "namespace" : "test",\n    "type" : "string"\n  } ],\n  "name" : "example",\n  "namespace" : "example",\n  "type" : "record"\n}',
        env: "DEV",
        showNext: false,
        showPrev: false,
        latest: true,
      },
    ],
  },
};

const mockUseTopicDetailsDataWithoutPromotion = {
  topicName: "aivendemotopic",
  environmentId: "1",
  topicOverview: {
    topicExists: true,
    schemaExists: false,
    topicInfo: {
      topicName: "aivendemotopic",
      noOfPartitions: 1,
      noOfReplicas: "1",
      teamname: "Ospo",
      teamId: 1003,
      envId: "1",
      showEditTopic: true,
      showDeleteTopic: false,
      topicDeletable: false,
      envName: "DEV",
    },
    aclInfoList: [
      {
        req_no: "1061",
        aclPatternType: "LITERAL",
        environment: "1",
        environmentName: "DEV",
        kafkaFlavorType: "AIVEN_FOR_APACHE_KAFKA",
        showDeleteAcl: true,
        teamid: 1003,
        teamname: "Ospo",
        topicname: "aivendemotopic",
        topictype: "Producer",
        acl_ssl: "aivendemotopicuser",
        consumergroup: "-na-",
      },
    ],
    topicHistoryList: [
      {
        environmentName: "DEV",
        teamName: "Ospo",
        requestedBy: "muralibasani",
        requestedTime: "2023-May-17 10:13:47",
        approvedBy: "josepprat",
        approvedTime: "2023-May-17 10:14:01",
        remarks: "Create",
      },
    ],
    availableEnvironments: [
      {
        id: "1",
        name: "DEV",
      },
      {
        id: "2",
        name: "TST",
      },
    ],
    topicPromotionDetails: {
      status: "SUCCESS",
      sourceEnv: "1",
      targetEnv: "TST",
      targetEnvId: "2",
      topicName: "aivendemotopic",
    },
  },
  topicSchemas: {
    topicExists: true,
    schemaExists: true,

    allSchemaVersions: {
      DEV: [1],
    },
    latestVersion: {
      DEV: 1,
    },
    schemaPromotionDetails: {
      DEV: {
        status: "NO_PROMOTION",
        sourceEnv: "3",
        targetEnv: "TST_SCH",
        targetEnvId: "9",
      },
    },
    schemaDetails: [
      {
        id: 2,
        version: 1,
        nextVersion: 0,
        prevVersion: 0,
        compatibility: "BACKWARD",
        content:
          '{\n  "doc" : "example",\n  "fields" : [ {\n    "default" : "6666665",\n    "doc" : "my test number",\n    "name" : "test",\n    "namespace" : "test",\n    "type" : "string"\n  } ],\n  "name" : "example",\n  "namespace" : "example",\n  "type" : "record"\n}',
        env: "DEV",
        showNext: false,
        showPrev: false,
        latest: true,
      },
    ],
  },
};
const mockClusterDetails: ClusterDetailsType = {
  allPageNos: [],
  currentPage: "1",
  bootstrapServers: "kafka-112233aa-dev-sandbeach.aivencloud.com:12345",
  clusterId: 999,
  clusterName: "DEV",
  clusterType: "Kafka",
  kafkaFlavor: "Aiven for Kafka",
  protocol: "SSL",
  showDeleteCluster: false,
  totalNoPages: "2",
  clusterStatus: "NOT_KNOWN",
};

describe("TopicOverview", () => {
  describe("with promotion banner", () => {
    let component: RenderResult;

    beforeAll(() => {
      mockedUseTopicDetails.mockReturnValue(
        mockUseTopicDetailsDataWithPromotion
      );
      mockGetClusterDetails.mockResolvedValue(mockClusterDetails);

      component = customRender(<TopicOverview />, {
        memoryRouter: true,
        queryClient: true,
      });
    });

    afterAll(() => {
      cleanup();
      jest.resetAllMocks();
    });

    it("renders correct DOM according to data from useTopicDetails and getClusterDetails", () => {
      expect(component.container).toMatchSnapshot();
    });

    it("renders the topic promotion banner", () => {
      const promotionBanner = screen.getByTestId("topic-promotion-banner");
      const promotionInfo = within(promotionBanner).getByText(
        "This topic has not yet been promoted to the TST environment."
      );

      expect(promotionBanner).toBeVisible();
      expect(promotionInfo).toBeVisible();
    });

    it("renders cluster details", () => {
      const clusterDefinitionList = getDefinitionList(component);

      expect(clusterDefinitionList).toBeVisible();
    });
  });

  describe("without promotion banner", () => {
    let component: RenderResult;
    beforeAll(() => {
      mockedUseTopicDetails.mockReturnValue(
        mockUseTopicDetailsDataWithoutPromotion
      );
      mockGetClusterDetails.mockResolvedValue(mockClusterDetails);

      component = customRender(<TopicOverview />, {
        memoryRouter: true,
        queryClient: true,
      });
    });

    afterAll(() => {
      cleanup();
      jest.clearAllMocks();
    });

    it("renders correct DOM according to data from useTopicDetails and getClusterDetails", () => {
      expect(component.container).toMatchSnapshot();
    });

    it("renders no topic promotion banner", () => {
      const promotionBanner = screen.queryByTestId("topic-promotion-banner");

      expect(promotionBanner).not.toBeInTheDocument();
    });

    it("renders cluster details", () => {
      const clusterDefinitionList = getDefinitionList(component);

      expect(clusterDefinitionList).toBeVisible();
    });
  });
});
