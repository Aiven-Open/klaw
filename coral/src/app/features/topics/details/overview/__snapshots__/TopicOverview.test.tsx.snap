// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicOverview with promotion banner renders correct DOM according to data from useTopicDetails and getClusterDetails 1`] = `
<div>
  <div
    style="display: grid; grid-template-columns: repeat(2, minmax(0, 1fr)); grid-template-rows: auto; gap: 24px;"
  >
    <div
      style="grid-column: [object Object];"
    >
      <div
        class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
      >
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="typography-default-strong text-black"
          >
            Topic details
          </div>
          <div
            class="typography-caption text-grey-70"
          >
            <div
              style="display: flex; gap: 160px;"
            >
              <div
                style="display: flex; justify-content: space-around; flex-direction: column;"
              >
                <div
                  class="typography-heading text-grey-70"
                >
                  1
                </div>
                <div
                  class="typography-small text-grey-50"
                >
                  Replicas
                </div>
              </div>
              <div
                style="display: flex; justify-content: space-around; flex-direction: column;"
              >
                <div
                  class="typography-heading text-grey-70"
                >
                  1
                </div>
                <div
                  class="typography-small text-grey-50"
                >
                  Partitions
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="grid-column: [object Object];"
    >
      <div
        data-testid="topic-promotion-banner"
      >
        <div
          class="Aquarium-Banner rounded flex justify-between gap-7 flex-nowrap p-6 bg-grey-0"
        >
          <div
            class="flex w-full flex-col"
          >
            <div
              class="typography-default text-grey-70 flex-grow"
            >
              <div
                style="display: flex; flex-direction: column; gap: 16px;"
              >
                <p
                  style="margin-bottom: 16px;"
                >
                  This 
                  topic
                   has not yet been promoted to the 
                  TST
                   environment.
                </p>
              </div>
              <a
                class="Aquarium-Button Aquarium-Button.Primary text-white bg-primary-80 active:bg-primary-90 active:ring-0 focus-visible:ring-2 focus-visible:ring-primary-100 focus-visible:ring-inset hover:bg-primary-90 disabled:bg-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 px-4 icon-stroke-2"
                href="/topic/aivendemotopic/request-promotion?sourceEnv=1&targetEnv=2"
              >
                Promote
              </a>
            </div>
          </div>
          <div
            style="display: inline-flex; align-items: center; margin: -24px -24px -24px 0px;"
          >
            <img
              alt=""
              class="bg-cover object-cover w-full"
              src="[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="grid-column: [object Object];"
    >
      <div
        class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
      >
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="typography-default-strong text-black"
          >
            Cluster details
          </div>
          <div
            class="typography-caption text-grey-70"
          >
            <div
              class="visually-hidden"
            >
              Cluster details are updating
            </div>
            <div
              aria-hidden="true"
            >
              <dl
                style="display: grid; grid-template-columns: repeat(2, minmax(0, 1fr)); row-gap: 24px;"
              >
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Bootstrap server
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Protocol
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Kafka flavor
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Rest API
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Cluster name
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Cluster status
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="typography-default-strong text-black"
        >
          Subscriptions
        </div>
        <div
          class="typography-caption text-grey-70"
        >
          <div
            style="display: flex; gap: 160px;"
          >
            <div
              style="display: flex; justify-content: space-around; flex-direction: column;"
            >
              <div
                class="typography-heading text-grey-70"
              >
                1
              </div>
              <div
                class="typography-small text-grey-50"
              >
                Producers
              </div>
            </div>
            <div
              style="display: flex; justify-content: space-around; flex-direction: column;"
            >
              <div
                class="typography-heading text-grey-70"
              >
                0
              </div>
              <div
                class="typography-small text-grey-50"
              >
                Consumers
              </div>
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; padding-top: 24px; gap: 32px;"
          >
            <a
              href="/topic/aivendemotopic/subscribe?env=1"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  style="display: flex; justify-content: center; align-items: center; flex-direction: row; gap: 8px;"
                >
                  <span
                    class="children:inline-block inline-flex justify-center items-center"
                  >
                    <svg
                      aria-hidden="true"
                      class="Aquarium-InlineIcon"
                      height="20px"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 22 22"
                      width="20px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <path
                        d="M11 7.333v7.334M7.333 11h7.334m5.5 0a9.167 9.167 0 11-18.334 0 9.167 9.167 0 0118.334 0Z"
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <div>
                    <div
                      class="typography-small-strong text-primary-80"
                    >
                      Request new subscription
                    </div>
                  </div>
                </div>
              </button>
            </a>
            <a
              href="/topic/aivendemotopic/subscriptions"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  class="typography-small-strong text-primary-80"
                >
                  See subscriptions
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="typography-default-strong text-black"
        >
          Schemas
        </div>
        <div
          class="typography-caption text-grey-70"
        >
          <div
            style="display: flex; justify-content: space-around; flex-direction: column;"
          >
            <div
              class="typography-heading text-grey-70"
            >
              1
            </div>
            <div
              class="typography-small text-grey-50"
            >
              Schemas
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; padding-top: 24px; gap: 32px;"
          >
            <a
              href="/topic/aivendemotopic/request-schema"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  style="display: flex; justify-content: center; align-items: center; flex-direction: row; gap: 8px;"
                >
                  <span
                    class="children:inline-block inline-flex justify-center items-center"
                  >
                    <svg
                      aria-hidden="true"
                      class="Aquarium-InlineIcon"
                      height="20px"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 22 22"
                      width="20px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <path
                        d="M11 7.333v7.334M7.333 11h7.334m5.5 0a9.167 9.167 0 11-18.334 0 9.167 9.167 0 0118.334 0Z"
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <div>
                    <div
                      class="typography-small-strong text-primary-80"
                    >
                      Request new schema
                    </div>
                  </div>
                </div>
              </button>
            </a>
            <a
              href="/topic/aivendemotopic/schema"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  class="typography-small-strong text-primary-80"
                >
                  See schema
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TopicOverview without promotion banner renders correct DOM according to data from useTopicDetails and getClusterDetails 1`] = `
<div>
  <div
    style="display: grid; grid-template-columns: repeat(2, minmax(0, 1fr)); grid-template-rows: auto; gap: 24px;"
  >
    <div
      style="grid-column: [object Object];"
    >
      <div
        class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
      >
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="typography-default-strong text-black"
          >
            Topic details
          </div>
          <div
            class="typography-caption text-grey-70"
          >
            <div
              style="display: flex; gap: 160px;"
            >
              <div
                style="display: flex; justify-content: space-around; flex-direction: column;"
              >
                <div
                  class="typography-heading text-grey-70"
                >
                  1
                </div>
                <div
                  class="typography-small text-grey-50"
                >
                  Replicas
                </div>
              </div>
              <div
                style="display: flex; justify-content: space-around; flex-direction: column;"
              >
                <div
                  class="typography-heading text-grey-70"
                >
                  1
                </div>
                <div
                  class="typography-small text-grey-50"
                >
                  Partitions
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="grid-column: [object Object];"
    >
      <div
        class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
      >
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="typography-default-strong text-black"
          >
            Cluster details
          </div>
          <div
            class="typography-caption text-grey-70"
          >
            <div
              class="visually-hidden"
            >
              Cluster details are updating
            </div>
            <div
              aria-hidden="true"
            >
              <dl
                style="display: grid; grid-template-columns: repeat(2, minmax(0, 1fr)); row-gap: 24px;"
              >
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Bootstrap server
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Protocol
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Kafka flavor
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Rest API
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Cluster name
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
                <div
                  style="display: flex; flex-direction: column;"
                >
                  <dt
                    class="inline-block mb-2 typography-small-strong text-grey-60"
                  >
                    Cluster status
                  </dt>
                  <dd
                    class="typography-small text-grey-70"
                  >
                    <div
                      aria-label="Loading..."
                      class="Aquarium-Skeleton bg-grey-5 h-auto before:content-['_'] whitespace-pre origin-[0%_45%] scale-[0.55] block animate-pulse"
                      role="progressbar"
                    />
                  </dd>
                </div>
              </dl>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="typography-default-strong text-black"
        >
          Subscriptions
        </div>
        <div
          class="typography-caption text-grey-70"
        >
          <div
            style="display: flex; gap: 160px;"
          >
            <div
              style="display: flex; justify-content: space-around; flex-direction: column;"
            >
              <div
                class="typography-heading text-grey-70"
              >
                1
              </div>
              <div
                class="typography-small text-grey-50"
              >
                Producers
              </div>
            </div>
            <div
              style="display: flex; justify-content: space-around; flex-direction: column;"
            >
              <div
                class="typography-heading text-grey-70"
              >
                0
              </div>
              <div
                class="typography-small text-grey-50"
              >
                Consumers
              </div>
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; padding-top: 24px; gap: 32px;"
          >
            <a
              href="/topic/aivendemotopic/subscribe?env=1"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  style="display: flex; justify-content: center; align-items: center; flex-direction: row; gap: 8px;"
                >
                  <span
                    class="children:inline-block inline-flex justify-center items-center"
                  >
                    <svg
                      aria-hidden="true"
                      class="Aquarium-InlineIcon"
                      height="20px"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 22 22"
                      width="20px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <path
                        d="M11 7.333v7.334M7.333 11h7.334m5.5 0a9.167 9.167 0 11-18.334 0 9.167 9.167 0 0118.334 0Z"
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <div>
                    <div
                      class="typography-small-strong text-primary-80"
                    >
                      Request new subscription
                    </div>
                  </div>
                </div>
              </button>
            </a>
            <a
              href="/topic/aivendemotopic/subscriptions"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  class="typography-small-strong text-primary-80"
                >
                  See subscriptions
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="border-grey-5 border-[1px] rounded-[2px] relative p-5 flex flex-col gap-5 w-full Aquarium-Card"
    >
      <div
        class="flex flex-col gap-4"
      >
        <div
          class="typography-default-strong text-black"
        >
          Schemas
        </div>
        <div
          class="typography-caption text-grey-70"
        >
          <div
            style="display: flex; justify-content: space-around; flex-direction: column;"
          >
            <div
              class="typography-heading text-grey-70"
            >
              1
            </div>
            <div
              class="typography-small text-grey-50"
            >
              Schemas
            </div>
          </div>
          <div
            style="display: flex; flex-direction: row; padding-top: 24px; gap: 32px;"
          >
            <a
              href="/topic/aivendemotopic/request-schema"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  style="display: flex; justify-content: center; align-items: center; flex-direction: row; gap: 8px;"
                >
                  <span
                    class="children:inline-block inline-flex justify-center items-center"
                  >
                    <svg
                      aria-hidden="true"
                      class="Aquarium-InlineIcon"
                      height="20px"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 22 22"
                      width="20px"
                      xmlns="http://www.w3.org/2000/svg"
                      xmlns:xlink="http://www.w3.org/1999/xlink"
                    >
                      <path
                        d="M11 7.333v7.334M7.333 11h7.334m5.5 0a9.167 9.167 0 11-18.334 0 9.167 9.167 0 0118.334 0Z"
                        fill="none"
                        stroke="currentColor"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                  <div>
                    <div
                      class="typography-small-strong text-primary-80"
                    >
                      Request new schema
                    </div>
                  </div>
                </div>
              </button>
            </a>
            <a
              href="/topic/aivendemotopic/schema"
            >
              <button
                class="Aquarium-Button Aquarium-Button.Ghost text-primary-80 active:text-primary-90 focus-visible:text-primary-90 hover:text-primary-90 disabled:text-primary-40 inline-block border-0 rounded-sm transition whitespace-nowrap focus:outline-none relative text-center typography-default-strong py-3 icon-stroke-2"
              >
                <div
                  class="typography-small-strong text-primary-80"
                >
                  See schema
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
